src/webhook/mutator.go

```diff
@@ -0,0 +1,45 @@
+package webhook
+
+import (
+	"context"
+
+	dynatracev1beta1 "github.com/Dynatrace/dynatrace-operator/src/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+type PodMutator interface {
+	Enabled(pod *corev1.Pod) bool
+	Injected(pod *corev1.Pod) bool
+	Mutate(request *MutationRequest) error
+	Reinvoke(request *ReinvocationRequest) bool
+}
+
+type MutationRequest struct {
```

Comment:
Why are struct's fileds pointers instead of values?

Reply:
Jet again because of the size,
also pod and installContainer should be mutated (the other 2 might make sense to not have pointers for)